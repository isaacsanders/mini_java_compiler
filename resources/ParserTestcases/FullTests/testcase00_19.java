class MainClass {
	public static void main(String[] a) {
		foo bar = new foo();
		while (!-!null/3 >= -true) {
			bar foo = this.baz(((-(0))));
		}
	}
}
