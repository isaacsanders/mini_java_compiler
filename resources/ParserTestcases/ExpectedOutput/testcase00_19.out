StmtList ::= 
Type ::= ID
CallExpr ::= NEW ID LPAREN RPAREN
UnaryExpr ::= CallExpr
Expr ::= UnaryExpr
Stmt ::= Type ID GETS Expr SEMI
StmtList ::= StmtList Stmt
LiteralExpr ::= NULL
CallExpr ::= LiteralExpr
UnaryExpr ::= CallExpr
UnaryExpr ::= NOT UnaryExpr
UnaryExpr ::= MINUS UnaryExpr
UnaryExpr ::= NOT UnaryExpr
Expr ::= UnaryExpr
LiteralExpr ::= INTEGER
CallExpr ::= LiteralExpr
UnaryExpr ::= CallExpr
Expr ::= UnaryExpr
Expr ::= Expr DIVIDE Expr
LiteralExpr ::= TRUE
CallExpr ::= LiteralExpr
UnaryExpr ::= CallExpr
UnaryExpr ::= MINUS UnaryExpr
Expr ::= UnaryExpr
Expr ::= Expr GTE Expr
StmtList ::= 
Type ::= ID
LiteralExpr ::= THIS
CallExpr ::= LiteralExpr
LiteralExpr ::= INTEGER
CallExpr ::= LiteralExpr
UnaryExpr ::= CallExpr
Expr ::= UnaryExpr
ParenExpr	::=	LPAREN Expr RPAREN
LiteralExpr ::= ParenExpr
CallExpr ::= LiteralExpr
UnaryExpr ::= CallExpr
UnaryExpr ::= MINUS UnaryExpr
Expr ::= UnaryExpr
ParenExpr	::=	LPAREN Expr RPAREN
LiteralExpr ::= ParenExpr
CallExpr ::= LiteralExpr
UnaryExpr ::= CallExpr
Expr ::= UnaryExpr
ParenExpr	::=	LPAREN Expr RPAREN
LiteralExpr ::= ParenExpr
CallExpr ::= LiteralExpr
UnaryExpr ::= CallExpr
Expr ::= UnaryExpr
ExprList ::= Expr
OptExprList ::= ExprList
CallExpr ::= CallExpr DOT ID LPAREN OptExprList RPAREN
UnaryExpr ::= CallExpr
Expr ::= UnaryExpr
Stmt ::= Type ID GETS Expr SEMI
StmtList ::= StmtList Stmt
Stmt ::= LBRACE StmtList RBRACE
Stmt ::= WHILE LPAREN Expr RPAREN Stmt
StmtList ::= StmtList Stmt
MainClassDecl ::= ...
ClassDeclList ::= 
Program ::= MainClassDecl ClassDeclList
